packages/dataviews/src/components/actions-toolbar/index.tsx

```diff
@@ -0,0 +1,205 @@
+			label={ label }
+			icon={ showLabels ? undefined : action.icon }
+			text={ showLabels ? label : undefined }
+			disabled={ !! action.disabled }
+			accessibleWhenDisabled
+			isDestructive={ action.isDestructive }
+			variant={ getVariant() }
+			size={ showLabels ? 'default' : 'compact' }
+			onClick={ onClick }
+		/>
+	);
+}
+
+function MenuItemTrigger< Item >( {
+	action,
+	onClick,
+	item,
+}: SingleItemActionTriggerProps< Item > ) {
+	const label =
+		typeof action.label === 'string'
+			? action.label
+			: action.label( [ item ] );
+	return (
+		<Menu.Item disabled={ action.disabled } onClick={ onClick }>
+			<Menu.ItemLabel>{ label }</Menu.ItemLabel>
+		</Menu.Item>
+	);
+}
+
+interface ActionsToolbarProps< Item > {
+	actions: Action< Item >[];
+	item: Item;
+	className?: string;
+	showLabels?: boolean;
+	variant?: 'toolbar' | 'menu';
+	size?: 'small' | 'compact';
+}
+
+/**
+ * A standalone toolbar component that renders DataViews actions as buttons or menu items.
+ *
+ * This component allows you to render DataViews actions inside and outside of the main DataViews component.
+ * It handles action eligibility, modal rendering, and maintains consistency with DataViews action patterns while being independently usable.
+ *
+ * @param props            - Component props
+ * @param props.actions    - Array of actions to render
+ * @param props.item       - Item the actions will operate on
+ * @param props.className  - Optional CSS class name
+ * @param props.showLabels - Whether to show text labels instead of icons (default: false, only applies to toolbar variant)
+ * @param props.variant    - Rendering style: 'toolbar' for horizontal buttons, 'menu' for dropdown (default: 'toolbar')
+ * @param props.size       - Button size: 'small', 'compact', or undefined for default (menu variant only)
+ */
+export default function ActionsToolbar< Item >( {
+	actions,
+	item,
+	className = '',
+	showLabels = false,
+	variant = 'toolbar',
+	size,
+}: ActionsToolbarProps< Item > ): ReactElement | null {
+	const registry = useRegistry();
+	const [ activeModalAction, setActiveModalAction ] =
+		useState< ActionModalType< Item > | null >( null );
+
+	// Filter actions that are eligible for this item
+	const eligibleActions = actions.filter( ( action ) => {
+		return ! action.isEligible || action.isEligible( item );
+	} );
+
+	if ( eligibleActions.length === 0 ) {
+		return null;
+	}
+
+	const handleActionClick = ( action: Action< Item > ) => {
+		if ( 'RenderModal' in action ) {
+			setActiveModalAction( action );
+			return;
+		}
+		action.callback( [ item ], { registry } );
+	};
+
+	if ( variant === 'menu' ) {
+		return (
+			<>
+				<Menu placement="bottom-end">
+					<Menu.TriggerButton
+						render={
+							<Button
+								size={ size }
+								icon={ moreVertical }
+								label={ __( 'Actions' ) }
+								accessibleWhenDisabled
+								disabled={ ! eligibleActions.length }
+								className={ `dataviews-all-actions-button ${ className }` }
```

Comment:
Probably shouldn't have this `dataviews-all-actions-button` classname here, I've taken it from the existing use, but it probably needs renaming / refactoring. It's not necessarily all of the actions for this item being passed here.